In [61]:
# hand written mean = 9.857

greg = 14
marcia = 12
peter = 11
jan = 10
bobby = 8
cindy = 6
cousin_o = 8 

mean_age = (greg + marcia + peter + jan+ bobby + cindy + cousin_o) / 7
print(mean_age)

9.857142857142858


In [62]:
import pandas as pd
import numpy as np

b_bunch = pd.DataFrame()

b_bunch['Kids'] = ['Greg', 'Marcia', 'Peter', ' Jan', 'Bobby', 'Cindy', 'Counsin Oliver']
b_bunch['Age'] = [14, 12, 11, 10, 8, 6, 8]

b_bunch

Kids  Age
0            Greg   14
1          Marcia   12
2           Peter   11
3             Jan   10
4           Bobby    8
5           Cindy    6
6  Counsin Oliver    8

In [63]:
b_bunch['Age'].mean()

9.857142857142858

In [64]:
# handwritten median = 10

In [65]:
b_bunch['Age'].median()

10.0

In [66]:
# handwritte mode = 8

In [67]:
b_bunch['Age'].mode()

0    8
dtype: int64

In [68]:
# handwritte var approx= 12.83 with use of calculator

In [69]:
b_bunch['Age'].var()

7.476190476190475

In [70]:
# my math was incorrect in the sum of squared error. 
# When I do it with a bias I get 6.41 but I do not understand why it is biased considering
# what I drew from was the entire population of starring children cast members. I assumed
# using n-1 was only when we drew a sample.


In [71]:
# handwritten std = 2.7

In [72]:
b_var = b_bunch['Age'].var()

b_var ** .5

2.734262327610589

In [73]:
b_bunch['Age'].std()

2.734262327610589

In [74]:
# approximate se = 1.023

In [75]:
(b_var ** .5) / (len(b_bunch['Age']) ** .5)

1.0334540197243192

In [76]:
b_bunch['Age'].std() / (len(b_bunch['Age']) ** .5)

1.0334540197243192

In [77]:
b_bunch.describe(include='all')

Kids        Age
count       7   7.000000
unique      7        NaN
top     Peter        NaN
freq        1        NaN
mean      NaN   9.857143
std       NaN   2.734262
min       NaN   6.000000
25%       NaN   8.000000
50%       NaN  10.000000
75%       NaN  11.500000
max       NaN  14.000000

Initially I would have chosen the mean, however, upon further inspection, I found that the median value extended by four values equally, while the mean was a little under.

my first assumption for the birthday of Cindy is that the mean will increase and the median and mode will stay the same. I also think the std will decrease. 

In [78]:
import pandas as pd
import numpy as np

b_bunch = pd.DataFrame()

b_bunch['Kids'] = ['Greg', 'Marcia', 'Peter', ' Jan', 'Bobby', 'Cindy', 'Counsin Oliver']
b_bunch['Age'] = [14, 12, 11, 10, 8, 7, 8]

b_bunch

Kids  Age
0            Greg   14
1          Marcia   12
2           Peter   11
3             Jan   10
4           Bobby    8
5           Cindy    7
6  Counsin Oliver    8

In [99]:
def test_stats(x, s):    
    print('mean: ', x[s].mean())
    print('median: ', x[s].median())
    print('mode: ', x[s].mode())
    print('var: ', x[s].var())
    print('std: ', x[s].std())
    print('se:', x[s].std() / (len(x[s]) ** .5))

    print('\n', b_bunch.describe())
    
test_stats(b_bunch, 'Age')

mean:  9.0
median:  10.0
mode:  0     1
1     7
2     8
3    10
4    11
5    12
6    14
dtype: int64
var:  18.0
std:  4.242640687119285
se: 1.6035674514745462

              Age
count   7.000000
mean    9.000000
std     4.242641
min     1.000000
25%     7.500000
50%    10.000000
75%    11.500000
max    14.000000


For the next question, having Cousin Oliver change to 1 year old jessica would greatly decrease the mean. median and mode will stay the same. I'm going to keep Cindy at 7.

In [80]:
b_bunch = pd.DataFrame()

b_bunch['Kids'] = ['Greg', 'Marcia', 'Peter', ' Jan', 'Bobby', 'Cindy', 'Jessica']
b_bunch['Age'] = [14, 12, 11, 10, 8, 7, 1]

b_bunch

Kids  Age
0     Greg   14
1   Marcia   12
2    Peter   11
3      Jan   10
4    Bobby    8
5    Cindy    7
6  Jessica    1

In [100]:
   
test_stats(b_bunch, 'Age')

mean:  9.0
median:  10.0
mode:  0     1
1     7
2     8
3    10
4    11
5    12
6    14
dtype: int64
var:  18.0
std:  4.242640687119285
se: 1.6035674514745462

              Age
count   7.000000
mean    9.000000
std     4.242641
min     1.000000
25%     7.500000
50%    10.000000
75%    11.500000
max    14.000000


I didn't realize that the second 8 would leave and thus return no mode. The mean didn't decrease by as dramatic an amount as I would have first thought. This leads me to believe I should have initially gone with the mean as my central tendency because it can change with the dataset. 

In [89]:
fans = pd.DataFrame()
fans['Magazine'] = ['Tv Guide', 'Entertainment Weekly', 'Pop Culture', 'SciPhi Phanatic']
fans['Fan poll'] = [20, 23, 17, 5]

fans

Magazine  Fan poll
0              Tv Guide        20
1  Entertainment Weekly        23
2           Pop Culture        17
3       SciPhi Phanatic         5

In [101]:
test_stats(fans, 'Fan poll')

mean:  16.25
median:  18.5
mode:  0     5
1    17
2    20
3    23
dtype: int64
var:  62.25
std:  7.88986691902975
se: 3.944933459514875

              Age
count   7.000000
mean    9.000000
std     4.242641
min     1.000000
25%     7.500000
50%    10.000000
75%    11.500000
max    14.000000


I would say that around 16% of people like The Brady Bunch